informer/src/handler/subscription.rs

```diff
@@ -14,28 +14,41 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{ColdEvents, EventTags, Events, Params, Sink};
+use crate::{ColdEvents, EventTags, Events, Params, Sink, SubscriptionId};
 use jsonrpc_core::futures::Future;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
 
 #[derive(Clone)]
 enum ConnectionState {
     Connected,
 }
 
-pub struct Connection {
+#[derive(Clone)]
+pub enum SubscriptionState {
+    Subscribed,
+    UnSubscribed,
+}
+
+#[derive(Clone)]
+pub struct Subscription {
     status: ConnectionState,
-    pub subscription_id: u64,
+    pub subscription_id: SubscriptionId,
     pub interested_events: Vec<EventTags>,
     sink: Sink,
+    pub cold_state: SubscriptionState,
+    pub state: Arc<AtomicBool>,
 }
 
-impl Connection {
-    pub fn new(sink: Sink, sub_id: u64) -> Self {
+impl Subscription {
+    pub fn new(sink: Sink, sub_id: SubscriptionId, cold_state: SubscriptionState) -> Self {
```

Comment:
`cold_event` -> `subscription_state`
```suggestion
    pub fn new(sink: Sink, sub_id: SubscriptionId, subscription_state: SubscriptionState) -> Self {
```